### PR TITLE
chore: run as non-root

### LINKED ISSUE #998: Run notebook service as non-root

### PR DESCRIPTION
**Pull request by olevski**
Ensures in the session and notebooks all containers run as non root.

closes #998 #999

/deploy #persist extra-values=notebooks.securityContext.enabled=null

### CODE REVIEW

File: git-https-proxy/Dockerfile [hunk 1]
```diff
@@ -5,4 +5,5 @@ RUN go build -o /git-http-proxy main.go
 
 FROM alpine:3.15
 COPY --from=builder /git-http-proxy /git-http-proxy
+USER 1000:100
```
Comment (Panaetius) [inline git-https-proxy/Dockerfile:8]:
```suggestion
USER 1000:1000
```
?

File: renku_notebooks/api/amalthea_patches/git_proxy.py [hunk 2]
```diff
@@ -20,6 +20,12 @@ def main(server):
                     "path": "/statefulset/spec/template/spec/containers/-",
                     "value": {
                         "image": current_app.config["GIT_HTTPS_PROXY_IMAGE"],
+                        "securityContext": {
+                            "fsGroup": 100,
+                            "runAsGroup": 100,
```
Comment (Panaetius) [inline renku_notebooks/api/amalthea_patches/git_proxy.py:25]: 1000?

**Review by Panaetius — COMMENTED** [review]
I think this is on purpose and not a typo, but shouldn't we be consistent with 1000:1000 ?

File: renku_notebooks/api/amalthea_patches/git_proxy.py (hunk 2 above)
Comment (olevski) [reply renku_notebooks/api/amalthea_patches/git_proxy.py:25]: The jupyterlab images run with fsgroup/group 100. Let me see if we can switch fully to 1000.

Comment (olevski) [reply renku_notebooks/api/amalthea_patches/git_proxy.py]: It works.

**Review by Panaetius — APPROVED** [review]

**Pull request merged**